serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
 ---
 title: Tools and Libraries
 author: clone95
-description: Learn to use existing libraries for Machine Learning and Deep Learning and out-of-the-box tools for productions systems.
----
\ No newline at end of file
+description: Learn to use existing libraries for Machine Learning and Deep Learning and out-of-the-box tools for production systems.
+---
+
+# Index
+- [How this guide is designed](#How-this-guide-is-designed)
+- [Solving the Learning Problem](#Solving-the-Learning-problem)
+- [Navigating the Ocean](#Navigating-the-Ocean)
+- [The Data Science Process Toolkit](#The-Data-Science-Process-Toolkit) 
+    + [Data Collection](#Data-Collection)
+    + [Data Preparation](#Data-Preparation)
+    + [Data Visualization](#Data-Visualization)
+    + [Machine Learning](#Machine-Learning)
+    + [Deep Learning](#Deep-Learning)
+    + [Hyperparameter Tuning](#Hyperparameter-Tuning)
+    + [Serving Models](#Serving-Models)
+    + [Miscellaneous](#Miscellaneous)
+
+- [Production Ready Tools](#Production-Ready-Tools)
+    + [Computer Vision](#Computer-vision)
+    + [Natural Language](#Natural-Language)
+    + [Time Series Forecasting](#Time-Series-Forecasting)
+
+- [Conclusions](#Conclusions)
+
+# How this guide is designed
+
+This guide is designed in the same fashion as the **Purgatorio** structure, so you find a subsection for each phase of the Data Science process.
+
+For _each section_, we list for you different libraries and the best resources to learn them.
+
+We'll give you some general tips to learn effectively and develop rock-solid foundations, that you can rely on to address and solve Data Science problems in the complexity of the real world (which is messy by definition).
+
+In this guide, you'll find a ton of different small tools, big libraries, and even entire frameworks, but be aware!
+
+As stated in the [Virgilio's Teaching Strategy](https://github.com/virgili0/Virgilio/blob/master/serving/paradiso/virgilio-teaching-strategy/virgilio-teaching-strategy.md) Guide, the best way to deal with the learning process is to leverage the "[Divide et Impera](https://en.wikipedia.org/wiki/Divide_and_rule)" principle, which states that every time you're tackling a problem (in this case, the _learning problem_) you should split it into smaller parts, easier to handle and understand. 
+
+# Solving the Learning Problem 
+
+The process of learning a tool and use it effectively **is equal to solving a problem**.
+
+We consider "solved" the problem when you can use the tool comfortably.
+
+So, you shouldn't try to learn all the features of a library in the same run, or you're going to be overwhelmed! 
+
+Every rich Python library contains thousands of different functions and utilities, but _you'll need a small subset to get the things done!_
+
+**So, Virgilio suggests the following approach**:
+
```

Comment:
The below advise is great although working on a project with well-defined high-level goals and then applying one or two libraries to meet those goals can be a good way to learn a library at the same time know how to implement it.

For e.g. doing a kaggle competition from scratch and then building your framework/pipeline and using libraries to solve the kaggle problem end-to-end can be a good learning exercise.

Reply:
True. This whole concept will have a dedicated guide in the section "Now go Build"

Reply:
Please link it here or make a footnote to the guide as it will be handy for them to refer when they come to this section.